lib/commands/lib/proxy_detect.js

```diff
@@ -0,0 +1,56 @@
+var exec = require('child_process').exec;
+
+/**
+ * Looks for proxy settings in some common places like ENV vars, Mac's networksetup
+ * @param {Function} callback - Function to run when a value has been determined
+ */
+exports.detect = function(callback) {
+	if (process.platform === 'darwin') {
+		['Ethernet', 'Wi-Fi'].forEach(function(svc) {
+			// while this runs for both interfaces, only one will typically be active
+			exec("networksetup -getsecurewebproxy \"" + svc + "\"", function (err, stdout, stderr) {
+				// if both http & https are configured, the https proxy is preferentially returned
+				if (stdout.indexOf("Enabled: Yes") !== -1) {
+					callback(parseNetSetup(stdout));
+				}
+			});
+			exec("networksetup -getwebproxy \"" + svc + "\"", function (err, stdout, stderr) {
+				if (stdout.indexOf("Enabled: Yes") !== -1) {
+					callback(parseNetSetup(stdout));
+				}
+			});
+		})
+	} else {
+		if (process.env['https_proxy'] != undefined) {
+			// if both configured, https proxy is preferentially returned
+			callback(parseEnv(process.env['https_proxy']));
+		} else if (process.env['http_proxy'] != undefined) {
+			callback(parseEnv(process.env['https_proxy']));
```

Comment:
if http_proxy is configured should the callback be to http_proxy?  (in the else if)